### ISSUE TITLE
LiteJet 48 and No Serial Comms

### ISSUE THREAD

**Issue by TopoTop1** [joncar/pylitejet#9]
Hi Jon,

Thank you so much for your help and offer to increase to LiteJet 48!! Here is a recap of your last email to refresh and the output you asked for:

`Yes, I can add that support if you can help by testing it. Can you create an issue in the https://github.com/joncar/pylitejet project? Let's start by figuring out why it doesn't like to show the first 24 loads which should be working as-is.
Please include the output of test:
git clone https://github.com/joncar/pylitejet
cd pylitejet
python3 ./test.py -v --path socket://<host>:<port> list -l -s -b
`

Output:

`SRSMServerNAS:/config/pylitejet# python3 ./test.py -v --path socket://<192.168.1.20>:<5000> list -l -s -b
bash: syntax error near unexpected token `5000'`

Where HOST = the Ip address of my Network Serial Device and PORT = the port number. 4999 is the port 1 and 5000 is port 2.

I have this device working via HA to connect to my MonoPrice 6 channel amplifier. It is connected to port 1 via port 4999 and HA is successfully controlling the amps. This test my socket connection as good.

I swapped the port around and tried to connect the LJ to port 4999 (known good working serial port from above) and I get the same results:

`SRSMServerNAS:/config/pylitejet# python3 ./test.py -v --path socket://<192.168.1.20>:<4999> list -l -s -b
bash: syntax error near unexpected token `4999'
SRSMServerNAS:/config/pylitejet#`

**Comment by joncar** [comment]
Ah. This might be the first problem. Your socket path syntax is wrong. That should be socket://192.168.1.20:4999

**Comment by TopoTop1** [comment]
> Ah. This might be the first problem. Your socket path syntax is wrong. That should be socket://192.168.1.20:4999

Thats cuz I am an idiot and wasn't paying attention!!! SORRY!

Correct output:

`SRSMServerNAS:/config/pylitejet# pylitejet python3 ./test.py -v --path socket://192.168.1.20:5000 list -l -s -b
bash: pylitejet: command not found
SRSMServerNAS:/config/pylitejet# `

**Comment by TopoTop1** [comment]
That's cuz I am an idiot and wasn't paying attention! SORRY.

Corrected output:

```
SRSMServerNAS:/config/pylitejet# python3 ./test.py -v --path socket://192.168.1.20:5000 list -l -s -b
DEBUG:pylitejet:Connecting to socket://192.168.1.20:5000
DEBUG:pylitejet:Connected to socket://192.168.1.20:5000
DEBUG:pylitejet:WantToSendRecv "^G"
DEBUG:pylitejet:SendRecv(S) "^G"
DEBUG:pylitejet:SendRecv(W)
INFO:pylitejet:No response to '^'. Trying '+'...
DEBUG:pylitejet:WantToSendRecv "+G"
DEBUG:pylitejet:SendRecv(S) "+G"
DEBUG:pylitejet:SendRecv(W)
INFO:pylitejet:Disconnecting by request
Cannot connect: No response to '+' or '^' command. No LiteJet MCP connected?
DEBUG:pylitejet:Connecting to socket://192.168.1.20:5000
DEBUG:pylitejet:Connected to socket://192.168.1.20:5000
```

The LJ is definitely connected to port 2 (the left one as looking at the MCP) of Panel 1. I know it is panel one as that is where the big display is connected to on the Master Port. 

I have ensured the 3rd party carriage return is "Enabled" in the software. It confirms the panel is set correctly.

The cable is known good as I can connect to the panel and program it via software.

**Comment by joncar** [comment]
Okay. Now we are getting somewhere. Does the same `^G` input give you a response over telnet? Might be that you need to set the option for "transmit a ‘CR’ after each response" in the programming software.

**Comment by TopoTop1** [comment]
Hi Jon,

I updated that post after I posted it whey this additional context that may
not have come through on email.

More details:

The LJ is definitely connected to port 2 (the left one as looking at the
MCP) of Panel 1. I know it is panel one as that is where the big display is
connected to on the Master Port.

I have ensured the 3rd party carriage return is "Enabled" in the software.
It confirms the panel is set correctly.

The cable is known good as I can connect to the panel and program it via
software.

**Comment by joncar** [comment]
Port 2 is on the right hand side (unless yours is mounted upside down). But looking at the install guide I don't think you can connect Home Assistant to port 2, because in a dual setup port 2 is how the two panels link together. So you have to connect to port 1. Connecting Home Assistant to port 1 should be fine, but you have to enable third party control on that port using the programming software:

> The LITEJET MCP can communicate with a third‐party device through two RS232 interfaces
RS232‐1 when DIP Switch # 8 is ON, and RS232‐2.

Looking ahead for when we get the single board working, I read through the dual MCP protocol PDFs. The one thing that is going to be complicated is I don't see a way for the MCP to report the load changed event for loads on the other board. I wonder if they were expecting a third party controller to connect to both MCPs. But they bothered to provide commands to *control* the other board. So, I don't know. We may have to experiment with that part.

**Comment by TopoTop1** [comment]
Ok, sounds interesting and I'm on board to figure this out.

I had on the link below:
[https://cocoontech.com/threads/centralite-litejet-and-omnipro-ii.24099/](url)

```
LiteJet 48 has two RS232 ports on the board in each box.  Port #2 connects box #1 to box #2 on each board.
 
You have to connect to the #1 RS232 port in box #1 to program both units...
```

Further in that discussion it says the below which I'll try in a few hours in case the programming blows up again the house isn't dark, that'd be bad WAF. The key appears to be the dip switch? They only mention dip switches when using firmware older than 5. Since I'm on 7 I skipped that. I'll also check on the protocols for the second board. 

`I ended up using RS232 port #1 on Unit #1 of my  LiteJet 48 to connect to my OmniPro II.  It doesn't direct you to switch DIP Switch #8 to on after programing the LiteJet via RS232 port #1 but you need to do this, and then push the reset button.  To skip having to do this every time it is recommend that you connect to OmniPro via RS232 Port #1 on Unit #2 as you can only program the LiteJet via RS232 Port #1 on Unit #1.  This will save you from having to switch the DIP Switch every time you need to reprogram`.

**Comment by TopoTop1** [comment]
Small update:

I changed dip switch 8 to ON on the MAIN panel and reran your script and got this output:
```

SRSMServerNAS:/config/pylitejet# python3 ./test.py -v --path socket://192.168.1.20:5000 list -l -s -b~
usage: Control a LiteJet lighting system. list [-h] [-l] [-s] [-b] [--all_switches] [--hide_off]
Control a LiteJet lighting system. list: error: argument -b/--buttons: ignored explicit argument '~'
SRSMServerNAS:/config/pylitejet# 
```

**Comment by joncar** [comment]
Copy and paste error. You've got a stray ~ at the end.

**Comment by TopoTop1** [comment]
OMG - I just for grins tried to set it up again and look what I got in HA!!!!

Are we in business????!!!! Please see attached screenshot. If we ARE in business, now we just need to look at supporting the 48. 

<img width="602" alt="LiteJet Success" src="https://github.com/joncar/pylitejet/assets/152359027/251e357e-e079-4c36-b8e4-fe2b419add93">

Also, when scrolling through the devices, it looks like it is not pulling the keypad names. Is this normal? It DOES appear to be pulling the light names, however.

This is very exciting and very promising!!!

Thank you for the patience as I work through this!

**Comment by TopoTop1** [comment]
Ok, re-ran without the stray and I got A WHOLE LOT returned!!! We are for sure connected now!!!

We should add some info to the instructions like this:

```
If you have Firmware version 5.00 or greater:

You will need to be connected to the LiteJet MCP using the LiteJet Programming Software in order to turn 3rd party CR on.  Once connected, go to the SETTINGS screen.  You will see a box with Baud Rate Settings for RS232-1 and RS232-2.  These are for the two RS232 ports on the front of the LiteJet MCP.   Click the "Current Settings" button to have it show you how it is set. You can select Baud Rates from 300 to 38,400 for each port AND turn on 3rd party CR. Click the "Send Settings" button ON THIS SETUP page, not the "Send" button at the top of the program.

CRITICAL STEP: You will need to set DIP Switch 8 to ON. This will put the panel in slave mode to accept the 3rd party control. 

You must Reset the MCP  by pressing the small black button below the DIP Switches after each such change for these Baud Rates to take effect.

** Note - You will need to flip DIP 8 to OFF and then reset the MCP as above to make any programing changes in the MCP. 

If your firmware is older than 5.00:

Use the DIP switches on the back of the MCP to set the Baud Rates.

DIP switch 3 controls the RS232-1 Baud Rate.  When DIP switch 3 is ON, RS232-1 Baud Rate is set to 9,600.  When DIP switch 3 is OFF, RS232-1 Baud Rate is set to 19,200.

DIP switch 4 controls the RS232-2 Baud Rate.  When DIP switch 4 is ON, RS232-2 Baud Rate is set to 9,600.  When DIP switch 4 is OFF, RS232-2 Baud Rate is set to 19,200.

RS232-1 only acts as a Third-Party port when DIP switch 8 is ON.  When DIP switch 8 is OFF, then RS232-1 acts as the programming port.

You must press the RESET button on the front of the MCP to make any of these changes take effect.
```

**Comment by joncar** [comment]
Nice!

Can you post a screenshot of what you mean by the keypad names? I think you mean the Home Assistant device list. In LiteJet each *button* can be named, but the *keypad overall* doesn't have a name so the Home Assistant LiteJet integration names them generically (but you can rename then in the Home Assistant UI, if you'd like).

**Comment by TopoTop1** [comment]
Ok,, Ok, it's official. I am now connected, and this is resolved as PILOT error. Final update about connectivity... These are the settings for LiteJet 24s and 48s regardless of SW rev. Set Panel 1 DIP Sw 8 to OFF, panel 2 dip 8 to ON (Or if you just running 1 24 panel, set dip 8 on the panel to ON). Reset both panels via the button right under the DIP switches.

The confusion for me was HAVING to use the dips as some of the documentation read like it was only for devices running FW older than Rev 5. Once I moved the dip, it just worked!

Ok, now on to the HA portion. I went ahead and loaded the devices to HA, they are now seen in there and their Status is shown as when imported during install. They are updating on their own and, when I click the toggle to switch the light on or off, it will respond correctly!!! 

I have HA connected to PANEL 2 in my 48 and it sees everything that is on panel 2, nothing from panel 1 - as expected. I then removed the devices, moved the connection to panel one and added the LJ again via HA and this time, it imported all of the lights and switches from Panel 1 and not 2, as expected. 

The button name is a NON-issue for me as I will not be using that string for anything yet. I was going to use for troubleshooting if  a button got stuck, I wanted to run a script or something but, that can totally wait. I can also just manually put it in if I ever do decide to use it.

I am going to delete this "Hub" in the litejet config so i can try it again when the update is ready for testing.

Thank you so much for your willingness to try this for me!!
